pkg/resources/management.cattle.io/v3/cluster/validator.go

```diff
@@ -265,6 +280,154 @@ func (a *admitter) validatePSACT(oldCluster, newCluster *apisv3.Cluster, op admi
+	}
+
+	if newPC.Preemption != nil && *newPC.Preemption != corev1.PreemptNever && *newPC.Preemption != corev1.PreemptLowerPriority && *newPC.Preemption != "" {
+		return admission.ResponseBadRequest("Priority Class Preemption value must be 'Never', 'PreemptLowerPriority', or empty")
+	}
+
+	if newPC.Value > 1000000000 {
+		return admission.ResponseBadRequest("Priority Class value cannot be greater than 1 billion")
+	}
+
+	if newPC.Value < -1000000000 {
+		return admission.ResponseBadRequest("Priority Class value cannot be less than negative 1 billion")
+	}
+
+	return admission.ResponseAllowed()
+}
+
+// validatePodDisruptionBudget validates that the Pod Disruption Budget defined in the cluster SchedulingCustomization field is properly
+// configured. The cluster-agent-scheduling-customization feature must be enabled to configure a Pod Disruption Budget, however an existing
+// Pod Disruption Budget may be deleted even if the feature is disabled.
+func (a *admitter) validatePodDisruptionBudget(oldCluster, newCluster *apisv3.Cluster, op admissionv1.Operation) *admissionv1.AdmissionResponse {
+	if op != admissionv1.Create && op != admissionv1.Update {
+		return admission.ResponseAllowed()
+	}
+	newClusterScheduling := getSchedulingCustomization(newCluster)
+	oldClusterScheduling := getSchedulingCustomization(oldCluster)
+
+	var newPDB, oldPDB *apisv3.PodDisruptionBudgetSpec
+	if newClusterScheduling != nil {
+		newPDB = newClusterScheduling.PodDisruptionBudget
+	}
+
+	if oldClusterScheduling != nil {
+		oldPDB = oldClusterScheduling.PodDisruptionBudget
+	}
+
+	if newPDB == nil {
+		return admission.ResponseAllowed()
+	}
+
+	featuredEnabled, err := a.featureCache.Get(common.SchedulingCustomizationFeatureName)
+	if err != nil {
+		return admission.ResponseBadRequest(fmt.Sprintf("failed to determine status of '%s' feature", common.SchedulingCustomizationFeatureName))
+	}
+
+	enabled := featuredEnabled.Status.Default
+	if featuredEnabled.Spec.Value != nil {
+		enabled = *featuredEnabled.Spec.Value
+	}
+
+	// if the feature is disabled then we should not permit any changes between the old and new clusters other than deletion
+	if !enabled && oldPDB != nil {
+		if reflect.DeepEqual(*oldPDB, *newPDB) {
+			return admission.ResponseAllowed()
+		}
+
+		return admission.ResponseBadRequest(fmt.Sprintf("'%s' feature is disabled, will only permit removal of Scheduling Customization fields until reenabled", common.SchedulingCustomizationFeatureName))
+	}
+
+	if !enabled && oldPDB == nil {
+		return admission.ResponseBadRequest(fmt.Sprintf("the '%s' feature must be enabled in order to configure a Priority Class or Pod Disruption Budget", common.SchedulingCustomizationFeatureName))
+	}
+
+	minAvailStr := newPDB.MinAvailable
+	maxUnavailStr := newPDB.MaxUnavailable
+
+	if (minAvailStr == "" && maxUnavailStr == "") ||
```

Comment:
Can you confirm that it is invalid to have both fields being empty?

Reply:
Correct, if the pod disruption budget field is provided, either `minAvailable` or `maxUnavailable` must be set to some value.